File: app/src/main/java/com/pizzamania/data/local/CartDao.java
package com.pizzamania.data.local;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import androidx.annotation.Nullable;
import kotlinx.coroutines.flow.Flow;

@Dao
public interface CartDao {

    @Query("SELECT * FROM cart_items WHERE branchId = :branchId ORDER BY localId DESC")
    Flow<List<CartItem>> observeCart(String branchId);

    // Same item + same options → merge qty
    @Query("SELECT * FROM cart_items " +
            "WHERE branchId = :branchId " +
            "AND itemId = :itemId " +
            "AND size = :size " +
            "AND COALESCE(extrasCsv,'') = COALESCE(:extrasCsv,'') " +
            "LIMIT 1")
    CartItem getItemWithOptions(String branchId, String itemId, String size, @Nullable String extrasCsv);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(CartItem item);

    @Update
    void update(CartItem item);

    @Delete
    void delete(CartItem item);

    @Query("DELETE FROM cart_items WHERE branchId = :branchId")
    void clearBranch(String branchId);
}

File: app/src/main/java/com/pizzamania/data/local/CartDao.kt
package com.pizzamania.data.local

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface CartDao {
    @Query("SELECT * FROM cart_items WHERE branchId = :branchId ORDER BY localId DESC")
    fun observeCart(branchId: String): Flow<List<CartItem>>

    @Query("SELECT * FROM cart_items WHERE branchId = :branchId AND itemId = :itemId LIMIT 1")
    suspend fun getItem(branchId: String, itemId: String): CartItem?

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(item: CartItem): Long

    @Update
    suspend fun update(item: CartItem)

    @Delete
    suspend fun delete(item: CartItem)

    @Query("DELETE FROM cart_items WHERE branchId = :branchId")
    suspend fun clearBranch(branchId: String)
}
